utils/swift_build_support/swift_build_support/toolchain.py

```diff
@@ -62,8 +63,12 @@ def _getter(self):
 _register("llvm_cov", "llvm-cov")
 _register("lipo", "lipo")
 _register("libtool", "libtool")
-_register("ranlib", "ranlib")
-_register("ar", "ar")
+if 'ANDROID_DATA' in os.environ:
+    _register("ranlib", "llvm-ranlib")
+    _register("ar", "llvm-ar")
```

Comment:
Honestly, this is just things needed for your specific environment. LLVM can install symlinks with the unprefixed name if you ask. You can create this symlinks in your machine. The LLVM reimplementations are supposed to be command-line compatible with the GNU counterparts most of the time.

Reply:
Yes and no, this checks if the `build-script` is running natively on Android itself, which means it's almost certainly running in the Termux app. With the switch to NDK 23b, I've switched both the native clang and swift toolchains in Termux to use llvm-ar directly, as that's what the NDK 23b does without providing a symlink to `ar`. This keeps everything consistent across the NDK and the Termux app.

Since it keeps everything consistent and only ever affects those few people building the Swift toolchain natively on Android, this shouldn't affect anybody else but makes native compilation on Android work out of the box.